both/collections/registration.js

```diff
@@ -16,25 +16,20 @@ Registrations.attachSchema({
   uniqueName: {
     type: String,
     custom: function() {
-      // this.docId is not available to the client on the first validation check
-      // due to a bug. When the server validation hits then the docId makes it
-      // back to the client.
+      // this.docId is not available to the client on the first validation check due to a
+      // bug. When the server validation hits then the docId makes it back to the client.
       // See: https://github.com/aldeed/meteor-collection2/pull/164
       // And: https://github.com/aldeed/meteor-simple-schema/issues/208
-      var docId = this.docId;
-      var userId = this.field('userId').value;
-      var compId = this.field('competitionId').value;
-      var uniqueName = this.value;
+      var obj = validationObject(this, ['uniqueName', 'userId', 'competitionId']);
 
       var uniqueMatch = Registrations.findOne({
-        competitionId: compId,
-        uniqueName: uniqueName,
+        competitionId: obj.competitionId,
+        uniqueName: obj.uniqueName,
       });
 
-      if(uniqueMatch && (this.isInsert || docId != uniqueMatch._id)) {
-        return "notUnique";
+      if(obj.uniqueMatch && (this.isInsert || obj.id != uniqueMatch._id)) {
+        return "notUnique"; // TODO this has no message?!?
```

Comment:
Running a competition with 2 people with the exact same name isn't really feasible, so we added a constraint that everyone at a competition have a `uniqueName`. There's a [database index](https://github.com/larspetrus/ccm/blob/master/both/collections/registration.js#L81-87) that insures this, all this code is doing is trying to get a nice error message to the user.